### DISCUSSION TITLE
Error when execute: invoke git-aggregate

### DISCUSSION THREAD

**Discussion by yasmanycastillo** [Tecnativa/doodba#495] [Doodba - The template] [answered]
Hi, when i create a new project using doodba-copier-template, when i execute invoke git-aggregate command i got this error.
The first time I tried to install everything was from my user, however I got the same error, so this time I tried to do it from the Odoo user that does have permissions on the /opt/odoo root, however as you can see I get the same error , How can i fix this?

```
odoo@yasmany:~/doodba$ invoke git-aggregate
Reinitialized existing Git repository in /opt/odoo/doodba/.git/
pre-commit installed at .git/hooks/pre-commit
doodba INFO: Running gitaggregate with /opt/odoo/auto/repos.yaml
(INFO) [00:42:21] git_aggregator.repo  odoo   Start aggregation of /opt/odoo/custom/src/odoo
(INFO) [00:42:21] git_aggregator.repo  odoo   Init empty git repository in /opt/odoo/custom/src/odoo
fatal: cannot mkdir /opt/odoo/custom/src/odoo: Permission denied
(ERROR) [00:42:21] git_aggregator.repo  odoo   /opt/odoo/custom/src/odoo> error calling ['git', 'init', '/opt/odoo/custom/src/odoo']
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/git_aggregator/main.py", line 219, in aggregate_repo
    repo.aggregate()
  File "/usr/local/lib/python3.8/site-packages/git_aggregator/repo.py", line 178, in aggregate
    self.init_repository(target_dir)
  File "/usr/local/lib/python3.8/site-packages/git_aggregator/repo.py", line 197, in init_repository
    self.log_call(['git', 'init', target_dir])
  File "/usr/local/lib/python3.8/site-packages/git_aggregator/repo.py", line 160, in log_call
    ret = callwith(cmd, **kw)
  File "/usr/local/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['git', 'init', '/opt/odoo/custom/src/odoo']' returned non-zero exit status 128.
Traceback (most recent call last):
  File "/usr/local/bin/autoaggregate", line 177, in <module>
    aggregate(AUTO_REPOS_YAML)
  File "/usr/local/bin/autoaggregate", line 72, in aggregate
    check_call(
  File "/usr/local/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['gitaggregate', '--expand-env', '--config', '/opt/odoo/auto/repos.yaml', '--log-level', 'INFO', '--jobs', '1', 'aggregate']' returned non-zero exit status 1.
```

**Comment by pedrobaeza** [comment]
Your folder doesn't have proper permissions. Are you working in a NTFS/FAT partition?

**Comment by yasmanycastillo** [comment]
No, I'm working on an ext4 partition, the opt/odoo path has all permissions. The path is the home of my odoo user, run everything with my regular user and the error persisted, try running it from the odoo user and it also persists.

**Reply by pedrobaeza** [reply]
Well, there is something incorrect with the permissions. I think you should start from scratch to see if fixed.

**Reply by yasmanycastillo** [reply]
Thanks for replying @pedrobaeza . I have already tried about 8 times without lying, doing everything from scratch with no different result. I have created the folders that the error mentions myself to see if it continued, I have changed the permissions and nothing at all.

**Reply by pedrobaeza** [reply]
Sorry for not being able to help, but I have never found such problem.

**Comment by bizzappdev** [comment]
This problem comes sometimes when you do not have proper write permission in odoo/auto folder. not sure why but somehow from some version of the copier or maybe a user error. it do not have write permission because the owner is root.
changing owner is solving the problem,

**Comment by yasmanycastillo** [comment, marked as answer]
I decided to close this question even though I couldn't fix it.
